src/werkzeug/datastructures/structures.pyi

```diff
@@ -74,6 +92,14 @@ class MultiDict(TypeConversionDict[K, V]):
     @overload
     def pop(self, key: K) -> V: ...
     @overload
+    def pop(self, key: K, default: V) -> V: ...
+    @overload
+    def pop(self, key: K, default: D) -> V | D: ...
+    @overload
+    def pop(self, key: K, default: D, type: Callable[[V], T]) -> D | T: ...
```

Comment:
This isn't valid unless the implementation of `MultiDict.pop` is changed to have a `type` parameter.

Similarly, `poplist` should probably have a `type` parameter.